### PR TITLE
Add powershell support to suppress UNC path error

### PR DESCRIPTION
**Pull request by spiffcs**
## Description

1) Prefer using `powershell.exe` over `cmd.exe` on wsl to supress terminal error:
`CMD.EXE was started with the above path as the current directory`

Follow up to issue #6338

Before:
image

After:
image

2) Add space for "-z" flag in help.fish. The new flag was causing the following error:
image

Previous change that caused the `missing close paren` error linked here
https://github.com/fish-shell/fish-shell/commit/ff2999ef2bc0b416e278f0cf8e975fd77457eaa8
  
## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: share/functions/help.fish [hunk 1]
```diff
@@ -86,6 +86,12 @@ function help --description 'Show help for the fish shell'
                 set fish_browser $cmd[1]
             end
 
+            # Try and see if powershell.exe is an option to avoid UNC path error
+            # https://ss64.com/nt/cmd.html Launching CMD/batch files from a UNC path
+            if type -q powershell.exe
+                set fish_browser powershell.exe
```
Comment (faho) [inline share/functions/help.fish:92]: This should be *inside* the WSL if above.

Make it 

```fish
and set -l cmd (command -s powershell.exe /mnt/c/Windows/System32/powershell.exe cmd.exe /mnt/c/Windows/System32/cmd.exe)
```

File: share/functions/help.fish [hunk 2]
```diff
@@ -224,10 +230,13 @@ function help --description 'Show help for the fish shell'
         end
     end
 
-    # cmd.exe needs more coaxing.
+    # cmd.exe and powershell needs more coaxing.
     if string match -qr 'cmd\.exe$' -- $fish_browser[1]
         # The space before the /c is to prevent msys2 from expanding it to a path
         $fish_browser " /c" start $page_url
+    else if string match -qr 'powershell\.exe$' -- $fish_browser[1]
```
Comment (faho) [inline share/functions/help.fish:237]: Just adjust the regex above - `if string match -qr 'powershell\.exe$|cmd\.exe$' -- $fish_browser[1]`

File: share/functions/help.fish (hunk 1 above)
Comment (spiffcs) [reply share/functions/help.fish]: updated conditional with the above `and` statement and set `fish_browser` as `$cmd[1]`

File: share/functions/help.fish (hunk 2 above)
Comment (spiffcs) [reply share/functions/help.fish]: adjusted regex with `if string match -qr 'powershell\.exe$|cmd\.exe$' -- $fish_browser[1]`

**Comment by spiffcs** [comment]
Thanks for the review @faho! 

Updates incorporated to the PR. 

Thanks also for all the fish-shell support.

File: share/functions/help.fish [hunk 3]
```diff
@@ -86,6 +86,14 @@ function help --description 'Show help for the fish shell'
                 set fish_browser $cmd[1]
             end
 
+            # Try and see if powershell.exe is an option to avoid UNC path error
+            # https://ss64.com/nt/cmd.html Launching CMD/batch files from a UNC path
+            if type -q powershell.exe
+                and set -l cmd (command -s powershell.exe cmd.exe /mnt/c/Windows/System32/cmd.exe)
```
Comment (faho) [inline share/functions/help.fish:92]: You misunderstood. I want the `powershell.exe` to go on line 84, inside the

```fish
if test -f /proc/version
   and string match -riq 'Microsoft|WSL|MSYS|MINGW' </proc/version
```

to ensure it only happens on WSL.

File: share/functions/help.fish [hunk 4]
```diff
@@ -86,6 +86,14 @@ function help --description 'Show help for the fish shell'
                 set fish_browser $cmd[1]
```
Comment (faho) [inline share/functions/help.fish:86]: ^^^ The line above this should simply include `powershell.exe`.

Comment (faho) [reply share/functions/help.fish:86]: (github doesn't allow me to put a comment there because it's too far from a changed part of the diff)

Comment (spiffcs) [reply share/functions/help.fish]: Ahh yea apologies for the misunderstanding here. 

This makes the diff much smaller and keeps everything where it should be - glossed over that block

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**